Doc/library/codecs.rst

```diff
@@ -697,8 +697,8 @@ compatible with the Python codec registry.
 
    .. method:: writelines(list)
 
-      Writes the concatenated list of strings to the stream (possibly by reusing
-      the :meth:`write` method). The standard bytes-to-bytes codecs
+      Writes the concatenated list of strings to the stream by reusing
+      the :meth:`write` method, and does not support infinite or very large generators. The standard bytes-to-bytes codecs
```

Comment:
This doesn't read well. Should put it in a separate sentence instead.

Reply:
How does this sound instead?

```
Writes the concatenated list of strings to the stream. Reuses the write method, so infinite or very large generators are not supported. The standard . . .
```

Reply:
Ah, that's a good point, then would this work?

```
Concatenates the string iterable and writes it to the stream. Uses .join() for concatenation, so infinite or very long iterables are not supported.
```

Reply:
That is correct, but it seems like the docstring goes out of its way to avoid saying too much about how the method is implemented, so maybe we should keep it less concrete.

Reply:
But the comment on not supporting large iterables is based off using `.join`, so wouldn't this be one of the cases where it'd be nice to go very briefly into implementation?

Reply:
I agree with Jelle about leaving out implementation detail because the `join()` wouldn't be an issue if it was buffering the lines, - so this can be confusing to readers. It can also be confusing to some readers because `join()` is recommended as the proper operation for concatenation as opposed to `+=`. You would have to go down the rabbit hole of explaining more and more details about implementation (which can also change in the future).

Reply:
Hmm interesting, so we would leave out the implementation, and if readers are curious on why infinite/large generators are not supported they would just read the source code themselves?

Reply:
Correct. The docstring here basically provides a guarantee to readers, so if we are too precise, that constrains how we can change the function in the future.